src/main/java/org/jmxtrans/agent/util/io/UrlResource.java

```diff
@@ -0,0 +1,178 @@
+        if (IoUtils.isFileUrl(url)) {
+            return new File(uri);
+        } else {
+            return super.getFile();
+        }
+    }
+
+    @Override
+    public boolean exists() {
+        if (IoUtils.isFileUrl(url)) {
+            return super.exists();
+        } else {
+            URLConnection conn;
+            try {
+                conn = url.openConnection();
+                if (conn instanceof HttpURLConnection) {
+                    HttpURLConnection httpConn = (HttpURLConnection) conn;
+                    configureUrlConnection(httpConn);
+                    try {
+                        int responseCode = httpConn.getResponseCode();
+                        if (responseCode == HttpURLConnection.HTTP_OK) {
+                            return true;
+                        } else if (responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
+                            return false;
+                        } else if (httpConn.getContentLength() > 0) {
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    } finally {
+                        httpConn.disconnect();
+                    }
+                } else {
+                    return super.exists();
+                }
+            } catch (IOException e) {
+                return false;
+            }
+        }
+    }
+
+    @Override
+    public long lastModified() throws IOException {
+        if (IoUtils.isFileUrl(url)) {
+            return super.lastModified();
+        } else {
+            URLConnection conn = null;
+            try {
+                conn = url.openConnection();
+                configureUrlConnection(conn);
+                if (conn instanceof HttpURLConnection) {
+                    HttpURLConnection httpConn = (HttpURLConnection) conn;
+                    int responseCode = httpConn.getResponseCode();
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        return httpConn.getLastModified();
+                    } else if (responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
+                        throw new FileNotFoundException(getDescription() + " not found: " +
+                                "responseCode=" + responseCode + ", usingProxy=" + httpConn.usingProxy());
+                    } else if (responseCode / 100 == 4) {
+                        throw new IOException(getDescription() + " is not accessible: " +
+                                "responseCode=" + responseCode + ", usingProxy=" + httpConn.usingProxy());
+                    } else if (responseCode / 100 == 5) {
+                        throw new IOException(getDescription() + " is not available: " +
+                                "responseCode=" + responseCode + ", usingProxy=" + httpConn.usingProxy());
+                    } else {
+                        return 0; // TODO should we fail?
```

Comment:
I see, didn't know that! If those are the cases we can get a 300, I am inclined towards throwing an error since I assume the lastModified in those cases are not really for the requested resouce.